### PR TITLE
Initial Base Class for a Generator

### PR DESCRIPTION
**Pull request by 1Maxnet1**
This PR introduces a base class for a generator. If you miss any attributes or functions t should offer, let me know.

### CODE REVIEW

**Review by EhrenbruderRoschtbraten — APPROVED** [review]

File: src/model/generator.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    """
+    Abstract Base Class of a Generator
+
+    This can be used as interface to abstract from different generating models
+    """
+
+    def __int__(self):
+        self.type = "abstract"
+
+    def generate(self):
```
Comment (EhrenbruderRoschtbraten) [inline src/model/generator.py:17]:
```suggestion
    def generate(self, output_file_path):
```

Don't we need an output path here, so its clear where to write the generated file.

**Review by EhrenbruderRoschtbraten — CHANGES_REQUESTED** [review]
I guess we need an output file path here.

File: src/model/generator.py (hunk 1 above)
Comment (LeviUniAcc) [reply src/model/generator.py:17]: i dont think we need it, because the user does not need to change it, so it can be static.

Comment (1Maxnet1) [reply src/model/generator.py:17]: Maybe let's just add a optional parameter, that allows to set it, but has a default path as well. What do you think?

**Comment by LeviUniAcc** [comment]
what does "enhanced" label mean? why does this issues has it?

**Comment by 1Maxnet1** [comment]
> what does "enhanced" label mean? why does this issues has it?

Nothing, removed it

**Review by LeviUniAcc — APPROVED** [review]

**Review by EhrenbruderRoschtbraten — APPROVED** [review]
Looks good

**Review by RaphaW — APPROVED** [review]

**Pull request merged**